File: 프로그래머스/unrated/135808. 과일 장수/과일 장수.swift
func solution(_ k:Int, _ m:Int, _ score:[Int]) -> Int {
    
    var dict : [String:Int] = [:]
    
    var maxPoint = k
    
    for i in score{
        if let j = dict[String(i)] {
            dict[String(i)]! += 1
        }else{
            dict[String(i)] = 1
        }
    }
    
    
    var cnt = 0
    var result = 0
    var boxCount = score.count / m
    
    
    while true{
        if let count = dict[String(maxPoint)] {
            if count > 0 {
                dict[String(maxPoint)]! -= 1
                cnt += 1
                if cnt == m {
                    result += cnt * maxPoint
                    boxCount -= 1
                    cnt = 0
                }
            }else{
                maxPoint -= 1
            }
        }else{
            maxPoint -= 1
        }


        if boxCount == 0{
            break
        }
    }

    return result
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.js
function solution(k, m, score) {
    let result = 0
    let appleBox = score.sort((a,b)=> b-a)
    for(let i=0; i<Math.floor(score.length/m); i++) {
        const box = []
        for(let j=m*i; j<(i+1)*m; j++) {
            box.push(appleBox[j])
        }
        result += Math.min(...box) * m
    }
    return result
}